File: android/src/main/kotlin/com/hodoan/ble_sdk/event/CharacteristicEvent.kt
package com.hodoan.ble_sdk.event

import android.os.Looper
import io.flutter.plugin.common.EventChannel

class CharacteristicEvent : EventChannel.StreamHandler {
    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }

    fun success(value: com.hodoan.ble_sdk.ProtobufModel.CharacteristicValue) {
        android.os.Handler(Looper.getMainLooper()).post { sink?.success(value.toByteArray()) }
    }
}

File: ios/Classes/Event/CharacteristicEvent.swift
//
//  CharacteristicEvent.swift
//  ble_sdk
//
//  Created by Doan Ho on 10/05/2023.
//

import Foundation
import Flutter

class CharacteristicEvent: NSObject, FlutterStreamHandler{
    private var sink:FlutterEventSink? = nil
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
    
    func sendData(_ data: CharacteristicValue) {
        sink?(try? data.serializedData())
        return
    }
}
